spring-vault-core/src/main/java/org/springframework/vault/authentication/MacAddressUserId.java

```diff
@@ -172,7 +173,7 @@ private static boolean matchesHint(String hint, NetworkInterface networkInterfac
 			return Optional.ofNullable(it.getHardwareAddress());
 		}
 		catch (SocketException e) {
-			throw new IllegalStateException(String
+			throw new VaultClientException(String
```

Comment:
`VaultException` here, too.

Reply:
Getting a hardware address and it failed isn't client/local specific?